docker/Dockerfile

```diff
@@ -0,0 +1,65 @@
+ARG ISAAC_SIM_VERSION=2023.1.1
+
+# Download base image
+FROM nvcr.io/nvidia/isaac-sim:${ISAAC_SIM_VERSION}
```

Comment:
Even though it is being pulled here I got this error:
```
Building the docker image for NVidia Isaac development.
[+] Building 0.8s (2/2) FINISHED                                                                                                                                                                                             docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                                                                   0.0s
 => => transferring dockerfile: 2.30kB                                                                                                                                                                                                 0.0s
 => ERROR [internal] load metadata for nvcr.io/nvidia/isaac-sim:2023.1.1                                                                                                                                                               0.8s
------
 > [internal] load metadata for nvcr.io/nvidia/isaac-sim:2023.1.1:
------
Dockerfile:4
--------------------
   2 |     
   3 |     # Download base image
   4 | >>> FROM nvcr.io/nvidia/isaac-sim:${ISAAC_SIM_VERSION}
   5 |     
   6 |     # Arguments for building
--------------------
ERROR: failed to solve: nvcr.io/nvidia/isaac-sim:2023.1.1: unexpected status from HEAD request to https://nvcr.io/v2/nvidia/isaac-sim/manifests/2023.1.1: 401 Unauthorized
```

So I ended up pulling the image (docker pull ...) before running ./build.sh script.   (yes, I am logged in into the nvcr) and that works correctly